ui/lib/core/addon/components/navigate-input.js

```diff
@@ -174,47 +179,46 @@ export default Component.extend(FocusOnInsertMixin, {
       });
     }
     this.transitionToRoute(...args);
-  },
+  }
 
-  filterUpdatedNoNav: function (val, mode) {
-    var key = val ? val.trim() : null;
-    this.transitionToRoute(routeFor('list-root', mode, this.urls), {
+  filterUpdatedNoNav(val, mode) {
+    const key = val ? val.trim() : null;
+    this.transitionToRoute(routeFor('list-root', mode, this.args.urls), {
       queryParams: {
         pageFilter: key,
         page: 1,
       },
     });
-  },
-
-  actions: {
-    handleInput: function (filter) {
-      if (this.filterDidChange) {
-        this.filterDidChange(filter);
-      }
-      debounce(this, 'filterUpdated', filter, 200);
-    },
-
-    setFilterFocused: function (isFocused) {
-      if (this.filterFocusDidChange) {
-        this.filterFocusDidChange(isFocused);
-      }
-    },
-
-    handleKeyPress: function (event) {
-      if (event.keyCode === keys.TAB) {
-        this.onTab(event);
-      }
-    },
+  }
 
-    handleKeyUp: function (event) {
-      var keyCode = event.keyCode;
-      const val = event.target.value;
-      if (keyCode === keys.ENTER) {
-        this.onEnter(val);
-      }
-      if (keyCode === keys.ESC) {
-        this.onEscape(val);
-      }
-    },
-  },
-});
+  @action
+  handleInput(filter) {
+    if (this.args.filterDidChange) {
+      this.args.filterDidChange(filter.target.value);
+    }
+    debounce(this, 'filterUpdated', filter.target.value, 200);
```

Comment:
it looks like the second arg here can also be `this.filterUpdated` since this arg can be a function or a string? [docs](https://api.emberjs.com/ember/release/functions/@ember%2Frunloop/debounce)

makes it a little easier to discern the second arg is a function, but I understand if you want to keep as is!